llvm/lib/Target/DirectX/DXILTranslateMetadata.cpp

```diff
@@ -103,6 +104,7 @@ class DXILTranslateMetadataLegacy : public ModulePass {
     AU.addRequired<DXILResourceWrapperPass>();
     AU.addRequired<DXILResourceMDWrapper>();
     AU.addRequired<ShaderFlagsAnalysisWrapper>();
+    AU.addRequired<DXILMetadataAnalysisWrapperPass>();
```

Comment:
Does `DXILMetadataAnalysisWrapperPass` need to be added as required for `DXILTranslateMetadata` pass, given the changes in this PR do not consume results from `DXILMetadataAnalysis` analysis pass?

Reply:
The analysis needs to be run before preparePass which removes the attributes.
And we could consume result for translateMetadata pass as well.